extensions/panache/mongodb-panache-rx/runtime/src/main/java/io/quarkus/mongodb/panache/rx/PanacheMongoEntityBase.java

```diff
@@ -0,0 +1,872 @@
+package io.quarkus.mongodb.panache.rx;
+
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletionStage;
+import java.util.stream.Stream;
+
+import org.bson.Document;
+import org.reactivestreams.Publisher;
+
+import io.quarkus.mongodb.ReactiveMongoCollection;
+import io.quarkus.mongodb.ReactiveMongoDatabase;
+import io.quarkus.mongodb.panache.rx.runtime.ReactiveMongoOperations;
+import io.quarkus.panache.common.Parameters;
+import io.quarkus.panache.common.Sort;
+import io.quarkus.panache.common.impl.GenerateBridge;
+
+/**
+ * Represents an entity. If your Mongo entities extend this class they gain auto-generated accessors
+ * to all their public fields, as well as a lot of useful
+ * methods. Unless you have a custom ID strategy, you should not extend this class directly but extend
+ * {@link PanacheMongoEntity} instead.
+ *
+ * @see PanacheMongoEntity
+ */
+public abstract class PanacheMongoEntityBase {
```

Comment:
This is not required but would make all non static method signature easier as they could use the T type parameter from the type instead of defining a type parameter for each method.

As Hibernate and Mongo panache extension didn't add a type type parameter I propose to not provide one here. This can be review later and added to the three extension without breaking compatibility.